sdks/python/apache_beam/examples/snippets/transforms/elementwise/filter_test.py

```diff
@@ -31,31 +31,26 @@
 
 
 def check_perennials(actual):
-  # [START perennials]
-  perennials = [
-      {'icon': '🍓', 'name': 'Strawberry', 'duration': 'perennial'},
-      {'icon': '🍆', 'name': 'Eggplant', 'duration': 'perennial'},
-      {'icon': '🥔', 'name': 'Potato', 'duration': 'perennial'},
-  ]
-  # [END perennials]
-  assert_that(actual, equal_to(perennials))
+  expected = '''[START perennials]
+{'icon': '🍓', 'name': 'Strawberry', 'duration': 'perennial'}
+{'icon': '🍆', 'name': 'Eggplant', 'duration': 'perennial'}
+{'icon': '🥔', 'name': 'Potato', 'duration': 'perennial'}
+[END perennials]'''.splitlines()[1:-1]
```

Comment:
I am confused why [START] and [END] tags needs to be part of this expected value? The list before the change looks simpler when reading the code.

Reply:
It is a little weird in the code, but it shows a lot nicer in the docs this way. That's the actual output people will get when running the code, rather than the lists.

The [START] and [END] tags need to be as part of the string so we can only extract the desired output, but they are being filtered out with the `.splitlines()[1:-1]`.

Reply:
What exactly users see will see when they run? And what exactly they will see in docs?

I think it will be confusing to the read of this code to have our doc tags mixed with code.

Reply:
If they run the code in the Colab notebook, or they copy/paste the code snippet into a file and run it, they will see the outputs of the `print` statements in `stdout`.

The docs are currently like this:

```py
import apache_beam as beam

with beam.Pipeline() as pipeline:
  perennials = (
      pipeline
      | 'Gardening plants' >> beam.Create([
          {'icon': '🍓', 'name': 'Strawberry', 'duration': 'perennial'},
          {'icon': '🥕', 'name': 'Carrot', 'duration': 'biennial'},
          {'icon': '🍆', 'name': 'Eggplant', 'duration': 'perennial'},
          {'icon': '🍅', 'name': 'Tomato', 'duration': 'annual'},
          {'icon': '🥔', 'name': 'Potato', 'duration': 'perennial'},
      ])
      | 'Filter perennials' >> beam.Filter(
          lambda plant: plant['duration'] == 'perennial')
      | beam.Map(print)
  )
```

Outputs:

```
perennials = [
    {'icon': '🍓', 'name': 'Strawberry', 'duration': 'perennial'},
    {'icon': '🍆', 'name': 'Eggplant', 'duration': 'perennial'},
    {'icon': '🥔', 'name': 'Potato', 'duration': 'perennial'},
]
```

But if they run the code, they'll actually see this:

```
{'icon': '🍓', 'name': 'Strawberry', 'duration': 'perennial'}
{'icon': '🍆', 'name': 'Eggplant', 'duration': 'perennial'}
{'icon': '🥔', 'name': 'Potato', 'duration': 'perennial'}
```

It's not much of a difference content-wise, but the format implies that it results in a list, when it in fact results in a PCollection, where we're only printing the values.

Also, the variable `perennials` by the end of the snippet is just a `PCollection` of `None` values since that's the output of `print`, so it's also misleading that way.

Furthermore, running the code in Colab gives the second results, while we see the first results in the docs, making them inconsistent.

Reply:
This make sense.